micro_sam/training/sam_trainer.py

```diff
@@ -242,11 +244,17 @@ def _update_masks(self, batched_inputs, y, sampled_binary_y, sampled_ids, num_su
         return loss, mask_loss, iou_regression_loss, mean_model_iou
 
     def _get_updated_points_per_mask_per_subiter(self, masks, sampled_binary_y, batched_inputs, logits_masks):
+        # function to adjust the samples based on the longest side resizing
+        transform = ResizeLongestSide(self.model.sam.image_encoder.img_size)
+
         # here, we get the pair-per-batch of predicted and true elements (and also the "batched_inputs")
         for x1, x2, _inp, logits in zip(masks, sampled_binary_y, batched_inputs, logits_masks):
             # here, we get each object in the pairs and do the point choices per-object
             net_coords, net_labels, _, _ = self.prompt_generator(x2, x1)
 
+            # convert the point coordinates to the expected resolution for iterative prompting
+            net_coords = transform.apply_coords_torch(net_coords, sampled_binary_y.shape[-2:])
+
```

Comment:
You probably don't need to update the masks, since they are always passed as lowres (256, 256) (I forgot about this earlier). But I think it's a good idea to leave a comment on this.